### PR TITLE
Dialog changes for new itwinui-react patch

### PR DESCRIPTION
**Pull request by AntanasBuk**
## Changes

- Upgraded to new itwinui-react patch version (2.11.6 -> 2.11.11) that fixes issues related to itwinui-react Dialog resizing.
- Adapted core-react Dialog to use resize functionality from itwinui-react Dialog.
- Removed unit tests related to Dialog resizing and dragging since this functionality comes from itwinui-react Dialog.

## Testing

Confirmed correct Dialog resizing behavior in test-app.

### CODE REVIEW

File: docs/changehistory/NextVersion.md [hunk 1]
```diff
@@ -45,6 +46,7 @@ Table of contents:
 ### Additions
 
 - `ConditionalIconItem.isConditionalIconItem` is now a `ConditionalIconItem` _type predicate_, on top of returning a boolean.
+- `Dialog` now has resize functionality from iTwinUI `Dialog`.
```
Comment (GerardasB) [inline docs/changehistory/NextVersion.md:49]: Move under `### Changes`. Reword to: "Changed `Dialog` to use resize..."?

File: ui/core-react/src/core-react/dialog/Dialog.scss [hunk 2]
```diff
@@ -7,10 +7,7 @@
 .core-dialog {
   @include uicore-z-index(dialog);
 
-  > .core-dialog-main {
-    display: flex;
```
Comment (GerardasB) [inline ui/core-react/src/core-react/dialog/Dialog.scss:11]: Is this removal intentional? Since resize handles use `absolute` position.

File: ui/core-react/src/core-react/dialog/Dialog.tsx [hunk 3]
```diff
@@ -143,36 +140,15 @@ export class Dialog extends React.Component<DialogProps, DialogState> {
...
-    };
-  }
-
-  private getParentWindow() {
-    // istanbul ignore next
-    return this._parentDocument.defaultView ?? window;
   }
 
   public override componentWillUnmount(): void {
-    const parentWindow = this.getParentWindow();
```
Comment (GerardasB) [inline ui/core-react/src/core-react/dialog/Dialog.tsx:163]: Have you checked if dialog resizing continues to work in a popout window?

File: docs/changehistory/NextVersion.md (hunk 1 above)
Comment (AntanasBuk) [reply docs/changehistory/NextVersion.md]: Done.

File: ui/core-react/src/core-react/dialog/Dialog.tsx (hunk 3 above)
Comment (AntanasBuk) [reply ui/core-react/src/core-react/dialog/Dialog.tsx:163]: Works.

**Comment by raplemie** [comment]
@AntanasBuk I'll handle the variant update so we can merge and release to 4.4.0

**Review by raplemie — APPROVED** [review]
LGTM!
